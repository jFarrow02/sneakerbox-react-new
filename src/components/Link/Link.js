import React from 'react';
import './Link.scss';
import PropTypes from 'prop-types';

const Link = props=>{

    const target = props.newWindow ? '_blank' : '_self';
    return(
        <a
            aria-label={props.linkDescription}
            onClick={props.clickHandler}
            href={props.url}
            target={target}
        >
            {props.children}
        </a>
    )
}

export default Link;

Link.propTypes = {
    clickHandler: PropTypes.func,
    hoverHandler: PropTypes.func,
    linkDescription: PropTypes.string,
    newWindow: PropTypes.bool,
    url: PropTypes.string,
}

Link.defaultProps ={
    clickHandler: null,
    hoverHandler: null,
    linkDescription: '',
    newWindow: false,
    url: '#',
}