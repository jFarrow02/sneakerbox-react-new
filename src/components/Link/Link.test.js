import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<Link/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})