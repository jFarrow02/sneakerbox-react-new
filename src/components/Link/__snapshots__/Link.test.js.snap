// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<a
  aria-label=""
  href="#"
  onClick={null}
  target="_self"
/>
`;
